### PR TITLE
Add dra, drm functions in ntheory module

### PR DESCRIPTION
**Pull request by namannimmo10**
#### References to other Issues or PRs

similar to #16496

#### Release Notes

* ntheory
    *  `dra`  function returns the additive digital root of integer `n` which is a value < 10
        obtained by adding the digits of n together (and repeating with the sum) until a single digit is obtained.
    *  `drm`  function returns the multiplicative digital root of integer n which is a value < 10 obtained by multiplying the digits of n together (and repeating with the product) until a single digit is obtained.

### CODE REVIEW

File: sympy/ntheory/factor_.py [hunk 1]
```diff
@@ -2371,3 +2371,49 @@ def is_amicable(m, n):
...
+        while a:
+            a, d = divmod(a, 10)
+            s += d
+        a = s
+    return a
+
+
+def drm(n):
+    """Return the multiplicative digital root of integer n which is
+    a value < 10 obtained by mutiplying the digits of n together
```
Comment (smichr) [inline sympy/ntheory/factor_.py]:
```suggestion
    a value < 10 obtained by multiplying the digits of n together
```

File: sympy/ntheory/tests/test_factor_.py [hunk 2]
```diff
@@ -658,3 +658,18 @@ def test_is_amicable():
     assert is_amicable(173, 129) is False
     assert is_amicable(220, 284) is True
     assert is_amicable(8756, 8756) is False
+
+
+def test_dra():
+    assert dra(789) == 6
+    assert dra(1234) == 1
```
Comment (smichr) [inline sympy/ntheory/tests/test_factor_.py]:
```suggestion
    assert dra(1234) == dra(-1234) == 1
```

File: sympy/ntheory/tests/test_factor_.py [hunk 3]
```diff
@@ -658,3 +658,18 @@ def test_is_amicable():
...
+    assert dra(23456789) == 8
+    assert dra(1000) == 1
+
+
+def test_drm():
+    assert drm(345) == 0
+    assert drm(1234) == 8
```
Comment (smichr) [inline sympy/ntheory/tests/test_factor_.py]:
```suggestion
    assert drm(1234) == drm(-1234) == 8
```

File: sympy/ntheory/factor_.py (hunk 1 above)
Comment (namannimmo10) [reply sympy/ntheory/factor_.py]: Fixed.

**Comment by smichr** [comment]
This is similar to #16496 - why not contribute by adding a review to the code there (which has the option to do this in an arbitrary base)? It also looks like your `drm` is better but your `dra` is less efficient.

**Comment by namannimmo10** [comment]
Sure, that one looks much better and efficient. I can pick any other issue for now.  Thanks for reviewing this PR.

**Comment by namannimmo10** [comment]
> but your `dra` is less efficient.

Hi @smichr 
Now, `dra` function computes the result in O(1) time complexity. 

```
def dra(n):
    """Return the additive digital root of integer n which is
    a value < 10 obtained by adding the digits of n together
    (and repeating with the sum) until a single digit is obtained.
    e.g. 789 -> 7 + 8 + 9 = 24 -> 2 + 4 = 6

    Examples
    ========

    >>> from sympy.ntheory.factor_ import dra
    >>> dra(789)
    6
    """
    a = abs(as_int(n))
    if a == 0:
        return 0
    res = a % 9
    if res == 0:
        res += 9
    return res
```
-------------------------------------------------------------------------------------
```
>>> from sympy import *
>>> dra(23456789)
8
>>> dra(0)
0
>>> dra(789)
6
>>> dra(1234)
1
>>> dra(-1234)
1
>>> dra(1000)
1

```

**Comment by smichr** [comment]
The earlier PR is still in progress and is more general.

**Comment by namannimmo10** [comment]
Alright, I'll close this.

**Pull request closed without merging**